app/services/xapi/base_service.rb

```diff
@@ -0,0 +1,10 @@
+module Xapi
+  class BaseService
+    include RoutesResolvable
+    attr_reader :agent, :object, :verb, :statement
+
+    def persist
+      Xapi.post_statement(remote_lrs: $remote_lrs, statement: @statement)
```

Comment:
`$remote_lrs` why global here ?
Why not pass this in the constructor? This will be really hard to test without some mocks etc.